### PR TITLE
update(docs): update readme to explain what it is briefly

### LINKED ISSUE #3: have an initial version of the basic docs
/kind documentation
/assign

Let's have a simple README.md, CONTRIBUTION.md to describe this project.

### PR DESCRIPTION
**Pull request by sanposhiho**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

update docs to explain what it is briefly

#### Which issue(s) this PR fixes:

Fixes #3 

#### Does this PR introduce a user-facing change?

No.

### CODE REVIEW

**Comment by sanposhiho** [comment]
Sorry, opened it because I wanted to check the bot's behavior. Actually, it's WIP yet.

/hold

File: README.md [hunk 1]
```diff
@@ -1,6 +1,27 @@
...
+
+This project is composed of:
+- The SDK to build your wasm-ed plugin.
+- The scheduler plugin to load your wasm-ed plugin.
+
+## Motivation
+
+Nowadays, the scheduler can be extended via 
+- [recommended] add custom scheduler plugins based on [Scheduling Framework](https://kubernetes.io/docs/concepts/scheduling-eviction/scheduling-framework/).
+- configure [Extenders](https://github.com/kubernetes/design-proposals-archive/blob/main/scheduling/scheduler_extender.md)
```
[the remark this answers is not in the document]
Comment (kerthcet) [reply README.md:13]: Then it looks like 
```
Nowadays, the kube-scheduler can be extended via scheduler framework ..., but it requires extra non-trivial works

File: README.md [hunk 2]
```diff
@@ -1,6 +1,27 @@
...
+Although building scheduler plugins is the recommended way to extend,
+it requires non-trivial works: 
+When you want to integrate your plugins into scheduler,
+you need to re-build the scheduler with your plugins, replace the vanilla scheduler with it, 
+and keep doing them whenever you want to upgrade your cluster.
+
+In this project, we aim that users can use their own plugins by giving wasm binary to the scheduler,
+which frees users from the above tasks.
+
+And furthermore, users write their own scheduler plugins in any languages other than Golang as well.
```
Comment (kerthcet) [inline README.md:24]:
```suggestion
Furthermore, programming language is no longer the problem anymore, you can use whatever programming language you like to build your own scheduler plugin.
```

**Review by kerthcet — COMMENTED** [review]
I'm not a native speaker, but I still left some comments for reference.

File: README.md [hunk 3]
```diff
@@ -1,6 +1,27 @@
 # kube-scheduler-wasm-extension
 
- All the things to make the scheduler extendable with wasm.
+All the things to make the scheduler extendable with [wasm](https://webassembly.org/).
+
+This project is composed of:
```
Comment (codefromthecrypt) [inline README.md:5]:
```suggestion
This project contains everything needed to extend the scheduler:
```

File: README.md [hunk 4]
```diff
@@ -1,6 +1,27 @@
...
+- The SDK to build your wasm-ed plugin.
+- The scheduler plugin to load your wasm-ed plugin.
```
Comment (codefromthecrypt) [inline README.md:7]:
```suggestion
- The scheduler extension which loads and runs wasm plugins
```

**Review by codefromthecrypt — APPROVED** [review]
I reworked the intro using some hints from the well-reviewed dapr docs which are similar. hope it helps. 

https://github.com/dapr/docs/blob/f89c4df19783a68902f093e2637304ecd3a221d2/daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (sanposhiho) [reply README.md:13]: > we didn't say anything why we should not use it. Focus on scheduler framework is general enough to me.

sg.

**Comment by sanposhiho** [comment]
PTAL @kerthcet @codefromthecrypt

File: README.md [hunk 5]
```diff
@@ -1,6 +1,29 @@
...
+This project allows you to extend the kube-scheduler with custom scheduler plugin compiled to a Wasm
+binary. It works by embedding a WebAssembly runtime, [wazero](https://wazero.io), into the
+scheduler, and loading custom scheduler plugin via configuration.
+
+This project contains everything needed to extend the scheduler:
+- Documentation describing what type of actions are possible, e.g. `Filter`.
+- Language SDKs used to build scheduler plugins, compiled to wasm.
+- The scheduler plugin which loads and runs wasm plugins
+
+## Motivation
```
Comment (codefromthecrypt) [inline README.md:15]: here's a rephrase of your motivation in perhaps more natural English:

Today, you can extend the kube-scheduler, but it is non-trivial. To customize the scheduler means writing Go, and a non-trivial build process. Once you've built your scheduler, you have deployment and configuration work to have your cluster use it. This isn't one time work, as it needs to be redone on every upgrade of your cluster.

This project lowers that tension by making the default scheduler capable of loading custom plugins, compiled to WebAssembly (wasm). This removes the deployment burdens above. It also allows plugins to be written in languages besides Go.

Comment (sanposhiho) [reply README.md:15]: Great, I fixed based on your suggestion.

**Review by codefromthecrypt — APPROVED** [review]

**Comment by sanposhiho** [comment]
@kerthcet could you check this one again please?

**Pull request merged**